core/modules/indexers/tag-indexer.js

```diff
@@ -35,9 +35,27 @@ TagIndexer.prototype.rebuild = function() {
 };
 
 TagIndexer.prototype.update = function(updateDescriptor) {
-	$tw.utils.each(this.subIndexers,function(subIndexer) {
-		subIndexer.update(updateDescriptor);
-	});
+	var newTid = updateDescriptor.new.tiddler;
+	var oldTid = updateDescriptor.old.tiddler;
+	var noOldTid = oldTid === undefined;
+	var noNewTid = newTid === undefined;
+
+	// Index should be kept when there is no new tiddler
+	if (noNewTid) {
+		return;	// early
+	} //  -> or
+	// a) new tiddler has no tags && no old tiddler -> or
+	// b) new tiddler has no tags && old tiddler has no tags
+	var a=((newTid && newTid.fields && newTid.fields.tags === undefined) && noOldTid),
+		b=((newTid && newTid.fields && newTid.fields.tags === undefined) && (oldTid && oldTid.fields && oldTid.fields.tags === undefined));
+
+	if( a || b ) {
+			return;
+		} else {
```

Comment:
https://eslint.org/docs/latest/rules/no-else-return

I thought we already have setup eslint? Did you get the warning?

Reply:
I do not get any warning. The eslint configuration we discussed for pretty printing is not active yet.